pay-api/src/pay_api/models/payment.py

```diff
@@ -213,19 +226,24 @@ def search_purchase_history(cls,  # pylint:disable=too-many-arguments, too-many-
 
         if not return_all:
             # Add pagination
-            pagination = query.paginate(per_page=limit, page=page)
-            result, count = pagination.items, pagination.total
+            sub_query = query.with_entities(Invoice.id).\
```

Comment:
`query.paginate` wouldn't seem to work for me. I've had a similar issue with routing slips before.